{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Scripts\n",
    "\n",
    "## Project: Diversity and Evolutionary Dynamics of Spore Coat Proteins on Spore-forming species of Bacillales and *Bacillus*\n",
    "\n",
    "### By Henry Secaira\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. Software used\n",
    "    1. Ubuntu v18.10 \n",
    "    2. Conda v4.8.3. Install Conda: https://docs.conda.io/projects/conda/en/latest/user-guide/install/linux.html\n",
    "    3. Python v3.7.6\n",
    "    4. Biopython v1.76. Install Biopython in Conda: https://anaconda.org/anaconda/biopython "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Steps to extract genes from pBLAST results for each genome/species\n",
    "\n",
    "1. Check and save significant pBLAST results\n",
    "    - In this step, we will create a new file to save only the significant BLAST results (protein accesion id, sequence, accession title).\n",
    "2. Extract genes from genbank files (genomes) using Protein Accesion from significant BLAST results\n",
    "    - In this step, we will extract all the genes using protein acession id (step 1) from a genbank file and save them in a new file.\n",
    "3. Change headers id from extracted genes (optional but helpful in future steps)\n",
    "    - In this step, we can change each sequence id, from protein acession id to gene name. \n",
    "    \n",
    "## Steps to creat a database of the same gene from multiple genomes/species (used for \"Selection methods\")\n",
    "\n",
    "1. Extract the same gene from multiple files (i.e. species)\n",
    "    - Let suppose we extracted all the coat proteins for each member of the Pumilus group. Now, in this step, we will create a single file that contain only one gene from all the members of the group.\n",
    "2. Delete stop codons of a gene database\n",
    "    - In this step, we will create a new file gene database with its sequences without stop codons, since it is required by some methods (MEME, BUSTED, CodeML).\n",
    "    \n",
    "### **All files must be in the same location, unless you specify the location!**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "query: cgeA\n",
      "match: NZ_CP017786.1_prot_WP_008358173.1_3417 [locus_tag=BK049_RS17445] [db_xref=GeneID:46135376] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_008358831.1_462 [locus_tag=BK049_RS02265] [db_xref=GeneID:46132401] [protein=h \n",
      "\n",
      "\n",
      "query: cgeB\n",
      "match: NZ_CP017786.1_prot_WP_071169050.1_3416 [locus_tag=BK049_RS17440] [db_xref=GeneID:46135375] [protein= \n",
      "\n",
      "\n",
      "query: cgeC\n",
      "match: NZ_CP017786.1_prot_WP_008358174.1_3418 [locus_tag=BK049_RS17450] [db_xref=GeneID:46135377] [protein= \n",
      "\n",
      "\n",
      "query: cgeD\n",
      "match: NZ_CP017786.1_prot_WP_051009114.1_3419 [locus_tag=BK049_RS17455] [db_xref=GeneID:46135378] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_071169213.1_1876 [locus_tag=BK049_RS09735] [db_xref=GeneID:46133856] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_008361207.1_2206 [locus_tag=BK049_RS11390] [db_xref=GeneID:46134181] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_008357501.1_2094 [locus_tag=BK049_RS10825] [db_xref=GeneID:46134072] [protein= \n",
      "\n",
      "\n",
      "query: cgeE\n",
      "\n",
      "\n",
      "query: cotA\n",
      "match: NZ_CP017786.1_prot_WP_071167971.1_979 [locus_tag=BK049_RS04945] [db_xref=GeneID:46132924] [protein=m \n",
      "\n",
      "\n",
      "query: cotB\n",
      "\n",
      "\n",
      "query: cotC\n",
      "\n",
      "\n",
      "query: cotD\n",
      "match: NZ_CP017786.1_prot_WP_008356833.1_3363 [locus_tag=BK049_RS19155] [db_xref=GeneID:46135657] [protein= \n",
      "\n",
      "\n",
      "query: cotE\n",
      "match: NZ_CP017786.1_prot_WP_007499466.1_8 [gene=cotE] [locus_tag=BK049_RS00035] [db_xref=GeneID:46131957]  \n",
      "\n",
      "\n",
      "query: cotF\n",
      "match: NZ_CP017786.1_prot_WP_071167854.1_726 [locus_tag=BK049_RS03580] [db_xref=GeneID:46132657] [protein=s \n",
      "\n",
      "\n",
      "query: cotG\n",
      "\n",
      "\n",
      "query: cotH\n",
      "match: NZ_CP017786.1_prot_WP_008355162.1_584 [locus_tag=BK049_RS02880] [db_xref=GeneID:46132522] [protein=s \n",
      "\n",
      "\n",
      "query: cotI\n",
      "match: NZ_CP017786.1_prot_WP_071168655.1_2443 [gene=yutH] [locus_tag=BK049_RS12530] [db_xref=GeneID:4613440 \n",
      "\n",
      "\n",
      "query: cotJA\n",
      "\n",
      "\n",
      "query: cotJB\n",
      "\n",
      "\n",
      "query: cotJC\n",
      "match: NZ_CP017786.1_prot_WP_071167690.1_310 [locus_tag=BK049_RS01530] [db_xref=GeneID:46132256] [protein=m \n",
      "\n",
      "\n",
      "query: cotM\n",
      "match: NZ_CP017786.1_prot_WP_008357838.1_3564 [locus_tag=BK049_RS18155] [db_xref=GeneID:46135518] [protein= \n",
      "\n",
      "\n",
      "query: cotO\n",
      "match: NZ_CP017786.1_prot_WP_071167770.1_486 [locus_tag=BK049_RS02385] [db_xref=GeneID:46132425] [protein=s \n",
      "\n",
      "\n",
      "query: cotP\n",
      "match: NZ_CP017786.1_prot_WP_071167921.1_888 [locus_tag=BK049_RS04490] [db_xref=GeneID:46132836] [protein=H \n",
      "\n",
      "\n",
      "query: cotQ\n",
      "\n",
      "\n",
      "query: cotR\n",
      "\n",
      "\n",
      "query: cotS\n",
      "match: NZ_CP017786.1_prot_WP_071168655.1_2443 [gene=yutH] [locus_tag=BK049_RS12530] [db_xref=GeneID:4613440 \n",
      "\n",
      "\n",
      "query: cotSA\n",
      "match: NZ_CP017786.1_prot_WP_008356785.1_3337 [gene=bshA] [locus_tag=BK049_RS17045] [db_xref=GeneID:4613529 \n",
      "match: NZ_CP017786.1_prot_WP_071168535.1_2205 [locus_tag=BK049_RS11385] [db_xref=GeneID:46134180] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_071168495.1_2089 [locus_tag=BK049_RS10800] [db_xref=GeneID:46134067] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_071169222.1_2069 [locus_tag=BK049_RS10700] [db_xref=GeneID:46134047] [protein= \n",
      "\n",
      "\n",
      "query: cotT\n",
      "\n",
      "\n",
      "query: cotU\n",
      "\n",
      "\n",
      "query: cotV\n",
      "match: NZ_CP017786.1_prot_WP_071167767.1_481 [locus_tag=BK049_RS02360] [db_xref=GeneID:46132420] [protein=s \n",
      "match: NZ_CP017786.1_prot_WP_071167769.1_483 [locus_tag=BK049_RS02370] [db_xref=GeneID:46132422] [protein=s \n",
      "\n",
      "\n",
      "query: cotW\n",
      "match: NZ_CP017786.1_prot_WP_071167768.1_482 [locus_tag=BK049_RS02365] [db_xref=GeneID:46132421] [protein=s \n",
      "\n",
      "\n",
      "query: cotX\n",
      "match: NZ_CP017786.1_prot_WP_071167769.1_483 [locus_tag=BK049_RS02370] [db_xref=GeneID:46132422] [protein=s \n",
      "\n",
      "\n",
      "query: cotY\n",
      "match: NZ_CP017786.1_prot_WP_008355373.1_484 [locus_tag=BK049_RS02375] [db_xref=GeneID:46132423] [protein=s \n",
      "match: NZ_CP017786.1_prot_WP_008355371.1_485 [locus_tag=BK049_RS02380] [db_xref=GeneID:46132424] [protein=s \n",
      "\n",
      "\n",
      "query: cotZ\n",
      "match: NZ_CP017786.1_prot_WP_008355371.1_485 [locus_tag=BK049_RS02380] [db_xref=GeneID:46132424] [protein=s \n",
      "match: NZ_CP017786.1_prot_WP_008355373.1_484 [locus_tag=BK049_RS02375] [db_xref=GeneID:46132423] [protein=s \n",
      "\n",
      "\n",
      "query: coxA/yrbB\n",
      "match: NZ_CP017786.1_prot_WP_071168855.1_2898 [locus_tag=BK049_RS14880] [db_xref=GeneID:46134869] [protein= \n",
      "\n",
      "\n",
      "query: cwIJ\n",
      "match: NZ_CP017786.1_prot_WP_008360333.1_1306 [locus_tag=BK049_RS06640] [db_xref=GeneID:46133253] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_071169001.1_3288 [gene=sleB] [locus_tag=BK049_RS16800] [db_xref=GeneID:4613524 \n",
      "match: NZ_CP017786.1_prot_WP_071167705.1_346 [locus_tag=BK049_RS01695] [db_xref=GeneID:46132289] [protein=c \n",
      "\n",
      "\n",
      "query: gerPA\n",
      "match: NZ_CP017786.1_prot_WP_003211745.1_586 [locus_tag=BK049_RS02890] [db_xref=GeneID:46132524] [protein=s \n",
      "match: NZ_CP017786.1_prot_WP_003211468.1_591 [locus_tag=BK049_RS02915] [db_xref=GeneID:46132529] [protein=s \n",
      "match: NZ_CP017786.1_prot_WP_008357197.1_2484 [locus_tag=BK049_RS12725] [db_xref=GeneID:46134441] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_008357200.1_2483 [locus_tag=BK049_RS12720] [db_xref=GeneID:46134440] [protein= \n",
      "\n",
      "\n",
      "query: gerPB\n",
      "match: NZ_CP017786.1_prot_WP_008355157.1_587 [locus_tag=BK049_RS02895] [db_xref=GeneID:46132525] [protein=g \n",
      "\n",
      "\n",
      "query: gerPC\n",
      "match: NZ_CP017786.1_prot_WP_008355155.1_588 [locus_tag=BK049_RS02900] [db_xref=GeneID:46132526] [protein=g \n",
      "\n",
      "\n",
      "query: gerPD\n",
      "match: NZ_CP017786.1_prot_WP_003210893.1_589 [locus_tag=BK049_RS02905] [db_xref=GeneID:46132527] [protein=g \n",
      "\n",
      "\n",
      "query: gerPE\n",
      "match: NZ_CP017786.1_prot_WP_008355151.1_590 [locus_tag=BK049_RS02910] [db_xref=GeneID:46132528] [protein=s \n",
      "\n",
      "\n",
      "query: gerPF\n",
      "match: NZ_CP017786.1_prot_WP_003211468.1_591 [locus_tag=BK049_RS02915] [db_xref=GeneID:46132529] [protein=s \n",
      "match: NZ_CP017786.1_prot_WP_003211745.1_586 [locus_tag=BK049_RS02890] [db_xref=GeneID:46132524] [protein=s \n",
      "match: NZ_CP017786.1_prot_WP_008357197.1_2484 [locus_tag=BK049_RS12725] [db_xref=GeneID:46134441] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_008357200.1_2483 [locus_tag=BK049_RS12720] [db_xref=GeneID:46134440] [protein= \n",
      "\n",
      "\n",
      "query: gerQ\n",
      "match: NZ_CP017786.1_prot_WP_008360964.1_1874 [gene=gerQ] [locus_tag=BK049_RS09725] [db_xref=GeneID:4613385 \n",
      "\n",
      "\n",
      "query: gerR/ylbO\n",
      "match: NZ_CP017786.1_prot_WP_071167661.1_214 [gene=gerR] [locus_tag=BK049_RS01060] [db_xref=GeneID:46132162 \n",
      "match: NZ_CP017786.1_prot_WP_071168413.1_1904 [locus_tag=BK049_RS09875] [db_xref=GeneID:46133884] [protein= \n",
      "\n",
      "\n",
      "query: gerT\n",
      "match: NZ_CP017786.1_prot_WP_008360004.1_3438 [locus_tag=BK049_RS17550] [db_xref=GeneID:46135397] [protein= \n",
      "\n",
      "\n",
      "query: lipC\n",
      "match: NZ_CP017786.1_prot_WP_071168069.1_1160 [locus_tag=BK049_RS05910] [db_xref=GeneID:46133113] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_083381142.1_3406 [locus_tag=BK049_RS17390] [db_xref=GeneID:46135365] [protein= \n",
      "\n",
      "\n",
      "query: oxdD\n",
      "match: NZ_CP017786.1_prot_WP_071168649.1_2428 [locus_tag=BK049_RS12455] [db_xref=GeneID:46134388] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_008360620.1_786 [locus_tag=BK049_RS03975] [db_xref=GeneID:46132736] [protein=o \n",
      "match: NZ_CP017786.1_prot_WP_008361546.1_1268 [locus_tag=BK049_RS06455] [db_xref=GeneID:46133217] [protein= \n",
      "\n",
      "\n",
      "query: safA\n",
      "match: NZ_CP017786.1_prot_WP_008361014.1_2897 [gene=safA] [locus_tag=BK049_RS14875] [db_xref=GeneID:4613486 \n",
      "match: NZ_CP017786.1_prot_WP_008357345.1_3462 [locus_tag=BK049_RS17670] [db_xref=GeneID:46135421] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_008359873.1_699 [locus_tag=BK049_RS03445] [db_xref=GeneID:46132630] [protein=C \n",
      "match: NZ_CP017786.1_prot_WP_008359989.1_3446 [locus_tag=BK049_RS17590] [db_xref=GeneID:46135405] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_008359989.1_3446 [locus_tag=BK049_RS17590] [db_xref=GeneID:46135405] [protein= \n",
      "\n",
      "\n",
      "query: spoIVA\n",
      "match: NZ_CP017786.1_prot_WP_008356730.1_3303 [gene=spoIVA] [locus_tag=BK049_RS16875] [db_xref=GeneID:46135 \n",
      "\n",
      "\n",
      "query: spoVID\n",
      "match: NZ_CP017786.1_prot_WP_071168843.1_2870 [locus_tag=BK049_RS14730] [db_xref=GeneID:46134841] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_071168843.1_2870 [locus_tag=BK049_RS14730] [db_xref=GeneID:46134841] [protein= \n",
      "\n",
      "\n",
      "query: spoVM\n",
      "match: NZ_CP017786.1_prot_WP_008354550.1_136 [locus_tag=BK049_RS00675] [db_xref=GeneID:46132085] [protein=s \n",
      "\n",
      "\n",
      "query: spsB\n",
      "match: NZ_CP017786.1_prot_WP_071168397.1_1877 [locus_tag=BK049_RS09740] [db_xref=GeneID:46133857] [protein= \n",
      "\n",
      "\n",
      "query: spsI\n",
      "match: NZ_CP017786.1_prot_WP_071168402.1_1883 [locus_tag=BK049_RS09770] [db_xref=GeneID:46133863] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_071168488.1_2078 [gene=galU] [locus_tag=BK049_RS10745] [db_xref=GeneID:4613405 \n",
      "match: NZ_CP017786.1_prot_WP_008360534.1_3504 [gene=galU] [locus_tag=BK049_RS17875] [db_xref=GeneID:4613546 \n",
      "match: NZ_CP017786.1_prot_WP_008355852.1_898 [gene=rfbF] [locus_tag=BK049_RS04540] [db_xref=GeneID:46132846 \n",
      "\n",
      "\n",
      "query: sscA\n",
      "\n",
      "\n",
      "query: sspO/cotK\n",
      "match: NZ_CP017786.1_prot_WP_071169117.1_3562 [locus_tag=BK049_RS18145] [db_xref=GeneID:46135516] [protein= \n",
      "\n",
      "\n",
      "query: tasA/cotN\n",
      "match: NZ_CP017786.1_prot_WP_071168937.1_3116 [locus_tag=BK049_RS15960] [db_xref=GeneID:46135083] [protein= \n",
      "\n",
      "\n",
      "query: tgl\n",
      "match: NZ_CP017786.1_prot_WP_071168694.1_2539 [locus_tag=BK049_RS12995] [db_xref=GeneID:46134495] [protein= \n",
      "\n",
      "\n",
      "query: yaaH\n",
      "match: NZ_CP017786.1_prot_WP_071168258.1_1563 [locus_tag=BK049_RS08110] [db_xref=GeneID:46133538] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_008359873.1_699 [locus_tag=BK049_RS03445] [db_xref=GeneID:46132630] [protein=C \n",
      "match: NZ_CP017786.1_prot_WP_008357345.1_3462 [locus_tag=BK049_RS17670] [db_xref=GeneID:46135421] [protein= \n",
      "\n",
      "\n",
      "query: ydgA\n",
      "\n",
      "\n",
      "query: ydgB\n",
      "\n",
      "\n",
      "query: ydhD\n",
      "match: NZ_CP017786.1_prot_WP_071168258.1_1563 [locus_tag=BK049_RS08110] [db_xref=GeneID:46133538] [protein= \n",
      "\n",
      "\n",
      "query: yeeK\n",
      "\n",
      "\n",
      "query: yhaX\n",
      "match: NZ_CP017786.1_prot_WP_008361321.1_668 [locus_tag=BK049_RS03290] [db_xref=GeneID:46132601] [protein=C \n",
      "match: NZ_CP017786.1_prot_WP_008355235.1_550 [locus_tag=BK049_RS02705] [db_xref=GeneID:46132488] [protein=C \n",
      "match: NZ_CP017786.1_prot_WP_008357079.1_1168 [locus_tag=BK049_RS05950] [db_xref=GeneID:46133121] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_008355652.1_1718 [gene=yidA] [locus_tag=BK049_RS08960] [db_xref=GeneID:4613370 \n",
      "match: NZ_CP017786.1_prot_WP_071167674.1_265 [locus_tag=BK049_RS01310] [db_xref=GeneID:46132212] [protein=C \n",
      "match: NZ_CP017786.1_prot_WP_071168455.1_2021 [locus_tag=BK049_RS10470] [db_xref=GeneID:46134001] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_071168455.1_2021 [locus_tag=BK049_RS10470] [db_xref=GeneID:46134001] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_008354793.1_2049 [locus_tag=BK049_RS10600] [db_xref=GeneID:46134027] [protein= \n",
      "\n",
      "\n",
      "query: yhbaqueg\n",
      "match: NZ_CP017786.1_prot_WP_034739126.1_741 [gene=queG] [locus_tag=BK049_RS03655] [db_xref=GeneID:46132672 \n",
      "\n",
      "\n",
      "query: yhbB\n",
      "match: NZ_CP017786.1_prot_WP_071167858.1_740 [locus_tag=BK049_RS03650] [db_xref=GeneID:46132671] [protein=a \n",
      "\n",
      "\n",
      "query: yhcN\n",
      "match: NZ_CP017786.1_prot_WP_008358712.1_728 [locus_tag=BK049_RS03590] [db_xref=GeneID:46132659] [protein=Y \n",
      "match: NZ_CP017786.1_prot_WP_008354755.1_245 [locus_tag=BK049_RS01215] [db_xref=GeneID:46132193] [protein=Y \n",
      "\n",
      "\n",
      "query: yhcQ\n",
      "match: NZ_CP017786.1_prot_WP_071167854.1_726 [locus_tag=BK049_RS03580] [db_xref=GeneID:46132657] [protein=s \n",
      "\n",
      "\n",
      "query: yheC\n",
      "match: NZ_CP017786.1_prot_WP_008361316.1_671 [locus_tag=BK049_RS03305] [db_xref=GeneID:46132604] [protein=s \n",
      "match: NZ_CP017786.1_prot_WP_008361313.1_672 [locus_tag=BK049_RS03310] [db_xref=GeneID:46132605] [protein=Y \n",
      "\n",
      "\n",
      "query: yheD\n",
      "match: NZ_CP017786.1_prot_WP_008361313.1_672 [locus_tag=BK049_RS03310] [db_xref=GeneID:46132605] [protein=Y \n",
      "\n",
      "\n",
      "query: yhjQ\n",
      "\n",
      "\n",
      "query: yhjR\n",
      "match: NZ_CP017786.1_prot_WP_008355136.1_597 [locus_tag=BK049_RS02945] [db_xref=GeneID:46132535] [protein=f \n",
      "\n",
      "\n",
      "query: yisY\n",
      "match: NZ_CP017786.1_prot_WP_034738998.1_1325 [locus_tag=BK049_RS06755] [db_xref=GeneID:46133272] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_071168688.1_2522 [locus_tag=BK049_RS12910] [db_xref=GeneID:46134478] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_071168731.1_2609 [gene=menH] [locus_tag=BK049_RS13445] [db_xref=GeneID:4613458 \n",
      "\n",
      "\n",
      "query: yjcB\n",
      "\n",
      "\n",
      "query: yjqC\n",
      "match: NZ_CP017786.1_prot_WP_071167690.1_310 [locus_tag=BK049_RS01530] [db_xref=GeneID:46132256] [protein=m \n",
      "\n",
      "\n",
      "query: yjzB\n",
      "match: NZ_CP017786.1_prot_WP_008355274.1_530 [locus_tag=BK049_RS02605] [db_xref=GeneID:46132468] [protein=h \n",
      "\n",
      "\n",
      "query: yknT\n",
      "match: NZ_CP017786.1_prot_WP_071167685.1_293 [locus_tag=BK049_RS01445] [db_xref=GeneID:46132239] [protein=s \n",
      "\n",
      "\n",
      "query: ykvP\n",
      "match: NZ_CP017786.1_prot_WP_071169050.1_3416 [locus_tag=BK049_RS17440] [db_xref=GeneID:46135375] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_008359873.1_699 [locus_tag=BK049_RS03445] [db_xref=GeneID:46132630] [protein=C \n",
      "\n",
      "\n",
      "query: ykvQ\n",
      "match: NZ_CP017786.1_prot_WP_071168258.1_1563 [locus_tag=BK049_RS08110] [db_xref=GeneID:46133538] [protein= \n",
      "\n",
      "\n",
      "query: ykzQ\n",
      "match: NZ_CP017786.1_prot_WP_008359989.1_3446 [locus_tag=BK049_RS17590] [db_xref=GeneID:46135405] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_008359989.1_3446 [locus_tag=BK049_RS17590] [db_xref=GeneID:46135405] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_008359989.1_3446 [locus_tag=BK049_RS17590] [db_xref=GeneID:46135405] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_008359989.1_3446 [locus_tag=BK049_RS17590] [db_xref=GeneID:46135405] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_008357345.1_3462 [locus_tag=BK049_RS17670] [db_xref=GeneID:46135421] [protein= \n",
      "\n",
      "\n",
      "query: ylbD\n",
      "match: NZ_CP017786.1_prot_WP_008354722.1_227 [locus_tag=BK049_RS01125] [db_xref=GeneID:46132175] [protein=h \n",
      "\n",
      "\n",
      "query: ymaG\n",
      "\n",
      "\n",
      "query: yncD\n",
      "match: NZ_CP017786.1_prot_WP_071167760.1_461 [gene=alr] [locus_tag=BK049_RS02260] [db_xref=GeneID:46132400] \n",
      "match: NZ_CP017786.1_prot_WP_071168044.1_1113 [gene=alr] [locus_tag=BK049_RS05665] [db_xref=GeneID:46133066 \n",
      "\n",
      "\n",
      "query: ypeP\n",
      "match: NZ_CP017786.1_prot_WP_008358101.1_3381 [locus_tag=BK049_RS17265] [db_xref=GeneID:46135340] [protein= \n",
      "\n",
      "\n",
      "query: yppG\n",
      "match: NZ_CP017786.1_prot_WP_071169023.1_3356 [locus_tag=BK049_RS17140] [db_xref=GeneID:46135316] [protein= \n",
      "\n",
      "\n",
      "query: ypzA\n",
      "\n",
      "\n",
      "query: yraD\n",
      "\n",
      "\n",
      "query: yraF\n",
      "\n",
      "\n",
      "query: yraG\n",
      "\n",
      "\n",
      "query: yrbC\n",
      "match: NZ_CP017786.1_prot_WP_007496614.1_838 [locus_tag=BK049_RS04235] [db_xref=GeneID:46132787] [protein=Y \n",
      "\n",
      "\n",
      "query: ysnD\n",
      "\n",
      "\n",
      "query: ysxE\n",
      "match: NZ_CP017786.1_prot_WP_071168844.1_2871 [gene=ysxE] [locus_tag=BK049_RS14735] [db_xref=GeneID:4613484 \n",
      "\n",
      "\n",
      "query: ytdA\n",
      "match: NZ_CP017786.1_prot_WP_071168488.1_2078 [gene=galU] [locus_tag=BK049_RS10745] [db_xref=GeneID:4613405 \n",
      "match: NZ_CP017786.1_prot_WP_008360534.1_3504 [gene=galU] [locus_tag=BK049_RS17875] [db_xref=GeneID:4613546 \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "match: NZ_CP017786.1_prot_WP_071168402.1_1883 [locus_tag=BK049_RS09770] [db_xref=GeneID:46133863] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_008355852.1_898 [gene=rfbF] [locus_tag=BK049_RS04540] [db_xref=GeneID:46132846 \n",
      "\n",
      "\n",
      "query: ytxO\n",
      "\n",
      "\n",
      "query: yusN\n",
      "match: NZ_CP017786.1_prot_WP_008356662.1_2391 [locus_tag=BK049_RS12285] [db_xref=GeneID:46134357] [protein= \n",
      "\n",
      "\n",
      "query: yutH\n",
      "match: NZ_CP017786.1_prot_WP_071168655.1_2443 [gene=yutH] [locus_tag=BK049_RS12530] [db_xref=GeneID:4613440 \n",
      "\n",
      "\n",
      "query: yuzC\n",
      "match: NZ_CP017786.1_prot_WP_034738645.1_2490 [locus_tag=BK049_RS12755] [db_xref=GeneID:46134447] [protein= \n",
      "\n",
      "\n",
      "query: ywqH\n",
      "match: NZ_CP017786.1_prot_WP_008355538.1_1773 [locus_tag=BK049_RS09225] [db_xref=GeneID:46133754] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_071168360.1_1775 [locus_tag=BK049_RS09235] [db_xref=GeneID:46133756] [protein= \n",
      "match: NZ_CP017786.1_prot_WP_034740044.1_1892 [locus_tag=BK049_RS09815] [db_xref=GeneID:46133872] [protein= \n",
      "\n",
      "\n",
      "query: ywrJ\n",
      "\n",
      "\n",
      "query: yxeE\n",
      "\n",
      "\n",
      "query: yybI\n"
     ]
    }
   ],
   "source": [
    "# Steps to extract genes from pBLAST results for each genome/species\n",
    "\n",
    "# Step 1: Check and save significant pBLAST results\n",
    "# create a .txt or .csv (recommended) file with the accession number of each hit\n",
    "\n",
    "#Blast\n",
    "from Bio.Blast import NCBIXML\n",
    "import sys\n",
    "\n",
    "## Values for significant hits, according to Pearson, 2013 ##\n",
    "E_VALUE_THRESH = 0.001\n",
    "BITS_SCORE = 40\n",
    "\n",
    "#Create the file to store your significant hits\n",
    "filename = sys.argv[0] = \"result_Bacillus_xiamenensis_VV3_biopython.csv\"\n",
    "OUT_test = open(filename, 'w')\n",
    "#OUT_test.write(\"Query_accession\\tE-value\\tBit_score\\tQuery\\n\") #in case you want a header in the file\n",
    "\n",
    "for result in sys.argv[2:]:\n",
    "    result_handle = open(\"result_Bacillus_xiamenensis_VV3.xml\") #BLast output file, must be in .XML\n",
    "    for record in NCBIXML.parse(result_handle): \n",
    "        if record.alignments: \n",
    "            print(\"\\n\") \n",
    "            print(\"query: %s\" % record.query[:100]) #print the name of each query (gene/prot) \n",
    "            for align in record.alignments: \n",
    "                for hsp in align.hsps: \n",
    "                    if hsp.expect < E_VALUE_THRESH and hsp.bits > BITS_SCORE: #only check significant results\n",
    "                        print(\"match: %s \" % align.title[:100])\n",
    "                        #print(hsp.expect) #if you want to know the E-value\n",
    "                        #print(hsp.bits) #if you want to know the Bits score\n",
    "                        #print(align.accession) #if you want to know the protein accession\n",
    "                        #record.query[:100] is the name of the query in BLAST (i.e. the gene name) *important*\n",
    "                        OUT_test.write(align.accession + \"\\t\" + record.query[:100] + \"\\t\" + align.title[:100] + \"\\n\")\n",
    "                        \n",
    "                        \n",
    "OUT_test.close()\n",
    "                    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Name NZ_CP017786, 7569 features\n"
     ]
    }
   ],
   "source": [
    "# Steps to extract genes from pBLAST results for each genome/species\n",
    "\n",
    "# Step 2: Extract genes from genbank files (genomes) using protein accession from significant BLAST results\n",
    "# 1\n",
    "\n",
    "# Verify your genbank file, with the accession number\n",
    "# There must be only one record for each genbank file (i.e. we assume the genbank file has no contigs)\n",
    "\n",
    "from Bio import SeqIO\n",
    "\n",
    "genbank_file = \"Bacillus_xiamenensis_VV3.gb\"\n",
    "for gb_record in SeqIO.parse(open(genbank_file,\"r\"), \"genbank\") :\n",
    "    print(\"Name %s, %i features\" % (gb_record.name, len(gb_record.features)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Looking at: WP_008358173.1\n",
      "Looking at: WP_008358831.1\n",
      "Looking at: WP_071169050.1\n",
      "Looking at: WP_008358174.1\n",
      "Looking at: WP_051009114.1\n",
      "Looking at: WP_071169213.1\n",
      "Looking at: WP_008361207.1\n",
      "Looking at: WP_008357501.1\n",
      "Looking at: WP_071167971.1\n",
      "Looking at: WP_008356833.1\n",
      "Looking at: WP_007499466.1\n",
      "Looking at: WP_071167854.1\n",
      "Looking at: WP_008355162.1\n",
      "Looking at: WP_071168655.1\n",
      "Looking at: WP_071167690.1\n",
      "Looking at: WP_008357838.1\n",
      "Looking at: WP_071167770.1\n",
      "Looking at: WP_071167921.1\n",
      "Looking at: WP_071168655.1\n",
      "Looking at: WP_008356785.1\n",
      "Looking at: WP_071168535.1\n",
      "Looking at: WP_071168495.1\n",
      "Looking at: WP_071169222.1\n",
      "Looking at: WP_071167767.1\n",
      "Looking at: WP_071167769.1\n",
      "Looking at: WP_071167768.1\n",
      "Looking at: WP_071167769.1\n",
      "Looking at: WP_008355373.1\n",
      "Looking at: WP_008355371.1\n",
      "Looking at: WP_008355371.1\n",
      "Looking at: WP_008355373.1\n",
      "Looking at: WP_071168855.1\n",
      "Looking at: WP_008360333.1\n",
      "Looking at: WP_071169001.1\n",
      "Looking at: WP_071167705.1\n",
      "Looking at: WP_003211745.1\n",
      "Looking at: WP_003211468.1\n",
      "Looking at: WP_008357197.1\n",
      "Looking at: WP_008357200.1\n",
      "Looking at: WP_008355157.1\n",
      "Looking at: WP_008355155.1\n",
      "Looking at: WP_003210893.1\n",
      "Looking at: WP_008355151.1\n",
      "Looking at: WP_003211468.1\n",
      "Looking at: WP_003211745.1\n",
      "Looking at: WP_008357197.1\n",
      "Looking at: WP_008357200.1\n",
      "Looking at: WP_008360964.1\n",
      "Looking at: WP_071167661.1\n",
      "Looking at: WP_071168413.1\n",
      "Looking at: WP_008360004.1\n",
      "Looking at: WP_071168069.1\n",
      "Looking at: WP_083381142.1\n",
      "Looking at: WP_071168649.1\n",
      "Looking at: WP_008360620.1\n",
      "Looking at: WP_008361546.1\n",
      "Looking at: WP_008361014.1\n",
      "Looking at: WP_008357345.1\n",
      "Looking at: WP_008359873.1\n",
      "Looking at: WP_008359989.1\n",
      "Looking at: WP_008359989.1\n",
      "Looking at: WP_008356730.1\n",
      "Looking at: WP_071168843.1\n",
      "Looking at: WP_071168843.1\n",
      "Looking at: WP_008354550.1\n",
      "Looking at: WP_071168397.1\n",
      "Looking at: WP_071168402.1\n",
      "Looking at: WP_071168488.1\n",
      "Looking at: WP_008360534.1\n",
      "Looking at: WP_008355852.1\n",
      "Looking at: WP_071169117.1\n",
      "Looking at: WP_071168937.1\n",
      "Looking at: WP_071168694.1\n",
      "Looking at: WP_071168258.1\n",
      "Looking at: WP_008359873.1\n",
      "Looking at: WP_008357345.1\n",
      "Looking at: WP_071168258.1\n",
      "Looking at: WP_008361321.1\n",
      "Looking at: WP_008355235.1\n",
      "Looking at: WP_008357079.1\n",
      "Looking at: WP_008355652.1\n",
      "Looking at: WP_071167674.1\n",
      "Looking at: WP_071168455.1\n",
      "Looking at: WP_071168455.1\n",
      "Looking at: WP_008354793.1\n",
      "Looking at: WP_034739126.1\n",
      "Looking at: WP_071167858.1\n",
      "Looking at: WP_008358712.1\n",
      "Looking at: WP_008354755.1\n",
      "Looking at: WP_071167854.1\n",
      "Looking at: WP_008361316.1\n",
      "Looking at: WP_008361313.1\n",
      "Looking at: WP_008361313.1\n",
      "Looking at: WP_008355136.1\n",
      "Looking at: WP_034738998.1\n",
      "Looking at: WP_071168688.1\n",
      "Looking at: WP_071168731.1\n",
      "Looking at: WP_071167690.1\n",
      "Looking at: WP_008355274.1\n",
      "Looking at: WP_071167685.1\n",
      "Looking at: WP_071169050.1\n",
      "Looking at: WP_008359873.1\n",
      "Looking at: WP_071168258.1\n",
      "Looking at: WP_008359989.1\n",
      "Looking at: WP_008359989.1\n",
      "Looking at: WP_008359989.1\n",
      "Looking at: WP_008359989.1\n",
      "Looking at: WP_008357345.1\n",
      "Looking at: WP_008354722.1\n",
      "Looking at: WP_071167760.1\n",
      "Looking at: WP_071168044.1\n",
      "Looking at: WP_008358101.1\n",
      "Looking at: WP_071169023.1\n",
      "Looking at: WP_007496614.1\n",
      "Looking at: WP_071168844.1\n",
      "Looking at: WP_071168488.1\n",
      "Looking at: WP_008360534.1\n",
      "Looking at: WP_071168402.1\n",
      "Looking at: WP_008355852.1\n",
      "Looking at: WP_008356662.1\n",
      "Looking at: WP_071168655.1\n",
      "Looking at: WP_034738645.1\n",
      "Looking at: WP_008355538.1\n",
      "Looking at: WP_071168360.1\n",
      "Looking at: WP_034740044.1\n",
      "DoNe!\n"
     ]
    }
   ],
   "source": [
    "# Steps to extract genes from pBLAST results for each genome/species\n",
    "\n",
    "# Step 2: Extract genes from genbank files (genomes) using protein accession from significant BLAST results\n",
    "# 2\n",
    "\n",
    "# Genbank file must have nucleotide sequences, at the end of the file!\n",
    "\n",
    "from Bio import SeqIO\n",
    "    \n",
    "###\n",
    "        \n",
    "def get_cds_feature_with_qualifier_value(seq_record, name, value):\n",
    "    \"\"\"Function to look for CDS feature by annotation value in sequence record.\n",
    "    \n",
    "    e.g. You can use this for finding features by locus tag, gene ID, or protein ID.\n",
    "    \"\"\"\n",
    "    # Loop over the features\n",
    "    for feature in genome_record.features:\n",
    "        if feature.type == \"CDS\" and value in feature.qualifiers.get(name, []):\n",
    "            return feature\n",
    "    # Could not find it\n",
    "    return None\n",
    "\n",
    "\n",
    "genome_record = SeqIO.read(\"Bacillus_xiamenensis_VV3.gb\", \"genbank\") #open your genbank file\n",
    "\n",
    "#list of accession numer (each accession must be \"WP_000116925.1\", with .1 at the end)\n",
    "#the list can be coppied from the .csv file generated in step 1. \n",
    "prot_id = [\"WP_008358173.1\",\n",
    "\"WP_008358831.1\",\n",
    "\"WP_071169050.1\",\n",
    "\"WP_008358174.1\",\n",
    "\"WP_051009114.1\",\n",
    "\"WP_071169213.1\",\n",
    "\"WP_008361207.1\",\n",
    "\"WP_008357501.1\",\n",
    "\"WP_071167971.1\",\n",
    "\"WP_008356833.1\",\n",
    "\"WP_007499466.1\",\n",
    "\"WP_071167854.1\",\n",
    "\"WP_008355162.1\",\n",
    "\"WP_071168655.1\",\n",
    "\"WP_071167690.1\",\n",
    "\"WP_008357838.1\",\n",
    "\"WP_071167770.1\",\n",
    "\"WP_071167921.1\",\n",
    "\"WP_071168655.1\",\n",
    "\"WP_008356785.1\",\n",
    "\"WP_071168535.1\",\n",
    "\"WP_071168495.1\",\n",
    "\"WP_071169222.1\",\n",
    "\"WP_071167767.1\",\n",
    "\"WP_071167769.1\",\n",
    "\"WP_071167768.1\",\n",
    "\"WP_071167769.1\",\n",
    "\"WP_008355373.1\",\n",
    "\"WP_008355371.1\",\n",
    "\"WP_008355371.1\",\n",
    "\"WP_008355373.1\",\n",
    "\"WP_071168855.1\",\n",
    "\"WP_008360333.1\",\n",
    "\"WP_071169001.1\",\n",
    "\"WP_071167705.1\",\n",
    "\"WP_003211745.1\",\n",
    "\"WP_003211468.1\",\n",
    "\"WP_008357197.1\",\n",
    "\"WP_008357200.1\",\n",
    "\"WP_008355157.1\",\n",
    "\"WP_008355155.1\",\n",
    "\"WP_003210893.1\",\n",
    "\"WP_008355151.1\",\n",
    "\"WP_003211468.1\",\n",
    "\"WP_003211745.1\",\n",
    "\"WP_008357197.1\",\n",
    "\"WP_008357200.1\",\n",
    "\"WP_008360964.1\",\n",
    "\"WP_071167661.1\",\n",
    "\"WP_071168413.1\",\n",
    "\"WP_008360004.1\",\n",
    "\"WP_071168069.1\",\n",
    "\"WP_083381142.1\",\n",
    "\"WP_071168649.1\",\n",
    "\"WP_008360620.1\",\n",
    "\"WP_008361546.1\",\n",
    "\"WP_008361014.1\",\n",
    "\"WP_008357345.1\",\n",
    "\"WP_008359873.1\",\n",
    "\"WP_008359989.1\",\n",
    "\"WP_008359989.1\",\n",
    "\"WP_008356730.1\",\n",
    "\"WP_071168843.1\",\n",
    "\"WP_071168843.1\",\n",
    "\"WP_008354550.1\",\n",
    "\"WP_071168397.1\",\n",
    "\"WP_071168402.1\",\n",
    "\"WP_071168488.1\",\n",
    "\"WP_008360534.1\",\n",
    "\"WP_008355852.1\",\n",
    "\"WP_071169117.1\",\n",
    "\"WP_071168937.1\",\n",
    "\"WP_071168694.1\",\n",
    "\"WP_071168258.1\",\n",
    "\"WP_008359873.1\",\n",
    "\"WP_008357345.1\",\n",
    "\"WP_071168258.1\",\n",
    "\"WP_008361321.1\",\n",
    "\"WP_008355235.1\",\n",
    "\"WP_008357079.1\",\n",
    "\"WP_008355652.1\",\n",
    "\"WP_071167674.1\",\n",
    "\"WP_071168455.1\",\n",
    "\"WP_071168455.1\",\n",
    "\"WP_008354793.1\",\n",
    "\"WP_034739126.1\",\n",
    "\"WP_071167858.1\",\n",
    "\"WP_008358712.1\",\n",
    "\"WP_008354755.1\",\n",
    "\"WP_071167854.1\",\n",
    "\"WP_008361316.1\",\n",
    "\"WP_008361313.1\",\n",
    "\"WP_008361313.1\",\n",
    "\"WP_008355136.1\",\n",
    "\"WP_034738998.1\",\n",
    "\"WP_071168688.1\",\n",
    "\"WP_071168731.1\",\n",
    "\"WP_071167690.1\",\n",
    "\"WP_008355274.1\",\n",
    "\"WP_071167685.1\",\n",
    "\"WP_071169050.1\",\n",
    "\"WP_008359873.1\",\n",
    "\"WP_071168258.1\",\n",
    "\"WP_008359989.1\",\n",
    "\"WP_008359989.1\",\n",
    "\"WP_008359989.1\",\n",
    "\"WP_008359989.1\",\n",
    "\"WP_008357345.1\",\n",
    "\"WP_008354722.1\",\n",
    "\"WP_071167760.1\",\n",
    "\"WP_071168044.1\",\n",
    "\"WP_008358101.1\",\n",
    "\"WP_071169023.1\",\n",
    "\"WP_007496614.1\",\n",
    "\"WP_071168844.1\",\n",
    "\"WP_071168488.1\",\n",
    "\"WP_008360534.1\",\n",
    "\"WP_071168402.1\",\n",
    "\"WP_008355852.1\",\n",
    "\"WP_008356662.1\",\n",
    "\"WP_071168655.1\",\n",
    "\"WP_034738645.1\",\n",
    "\"WP_008355538.1\",\n",
    "\"WP_071168360.1\",\n",
    "\"WP_034740044.1\"]\n",
    "\n",
    "#save the output in a new file\n",
    "\n",
    "with open(\"Bacillus_xiamenensis_VV3_genes.fas\", \"w\") as nt_output: #create the output file\n",
    "    for p_id in prot_id: #iterate from the list of proteins_id\n",
    "        print(\"Looking at: \" + p_id)\n",
    "        cds_feature = get_cds_feature_with_qualifier_value(genome_record, \"protein_id\", p_id) #call the function to find the cds\n",
    "        gene_sequence = cds_feature.extract(genome_record.seq) #extract the sequence of the feature returned by the function\n",
    "        #all extracted sequences are from 5' to 3', biopython does that! \n",
    "        #print(\"CDS nucleotide sequence:\")\n",
    "        #print(gene_sequence)\n",
    "        nt_output.write(\">%s\\n%s\\n\" % (p_id, gene_sequence)) #save each sequence\n",
    "        \n",
    "print(\"DoNe!\")\n",
    "\n",
    "# Sometimes you can get an error, and the function does not work because the protein acession was changed.\n",
    "# So, use the following to test if the accession number works.\n",
    "\n",
    "#genome_record = SeqIO.read(\"test2.gb\", \"genbank\")\n",
    "#cds_feature = get_cds_feature_with_qualifier_value(genome_record, \"protein_id\", \"WP_000116925.1\")\n",
    "#print(cds_feature)\n",
    "#print(cds_feature.location)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Looking at: WP_008358173.1 replacing with: cgeA\n",
      "Looking at: WP_008358831.1 replacing with: cgeA\n",
      "Looking at: WP_071169050.1 replacing with: cgeB\n",
      "Looking at: WP_008358174.1 replacing with: cgeC\n",
      "Looking at: WP_051009114.1 replacing with: cgeD\n",
      "Looking at: WP_071169213.1 replacing with: cgeD\n",
      "Looking at: WP_008361207.1 replacing with: cgeD\n",
      "Looking at: WP_008357501.1 replacing with: cgeD\n",
      "Looking at: WP_071167971.1 replacing with: cotA\n",
      "Looking at: WP_008356833.1 replacing with: cotD\n",
      "Looking at: WP_007499466.1 replacing with: cotE\n",
      "Looking at: WP_071167854.1 replacing with: cotF\n",
      "Looking at: WP_008355162.1 replacing with: cotH\n",
      "Looking at: WP_071168655.1 replacing with: cotI\n",
      "Looking at: WP_071167690.1 replacing with: cotJC\n",
      "Looking at: WP_008357838.1 replacing with: cotM\n",
      "Looking at: WP_071167770.1 replacing with: cotO\n",
      "Looking at: WP_071167921.1 replacing with: cotP\n",
      "Looking at: WP_071168655.1 replacing with: cotS\n",
      "Looking at: WP_008356785.1 replacing with: cotSA\n",
      "Looking at: WP_071168535.1 replacing with: cotSA\n",
      "Looking at: WP_071168495.1 replacing with: cotSA\n",
      "Looking at: WP_071169222.1 replacing with: cotSA\n",
      "Looking at: WP_071167767.1 replacing with: cotV\n",
      "Looking at: WP_071167769.1 replacing with: cotV\n",
      "Looking at: WP_071167768.1 replacing with: cotW\n",
      "Looking at: WP_071167769.1 replacing with: cotX\n",
      "Looking at: WP_008355373.1 replacing with: cotY\n",
      "Looking at: WP_008355371.1 replacing with: cotY\n",
      "Looking at: WP_008355371.1 replacing with: cotZ\n",
      "Looking at: WP_008355373.1 replacing with: cotZ\n",
      "Looking at: WP_071168855.1 replacing with: coxA/yrbB\n",
      "Looking at: WP_008360333.1 replacing with: cwIJ\n",
      "Looking at: WP_071169001.1 replacing with: cwIJ\n",
      "Looking at: WP_071167705.1 replacing with: cwIJ\n",
      "Looking at: WP_003211745.1 replacing with: gerPA\n",
      "Looking at: WP_003211468.1 replacing with: gerPA\n",
      "Looking at: WP_008357197.1 replacing with: gerPA\n",
      "Looking at: WP_008357200.1 replacing with: gerPA\n",
      "Looking at: WP_008355157.1 replacing with: gerPB\n",
      "Looking at: WP_008355155.1 replacing with: gerPC\n",
      "Looking at: WP_003210893.1 replacing with: gerPD\n",
      "Looking at: WP_008355151.1 replacing with: gerPE\n",
      "Looking at: WP_003211468.1 replacing with: gerPF\n",
      "Looking at: WP_003211745.1 replacing with: gerPF\n",
      "Looking at: WP_008357197.1 replacing with: gerPF\n",
      "Looking at: WP_008357200.1 replacing with: gerPF\n",
      "Looking at: WP_008360964.1 replacing with: gerQ\n",
      "Looking at: WP_071167661.1 replacing with: gerR/ylbO\n",
      "Looking at: WP_071168413.1 replacing with: gerR/ylbO\n",
      "Looking at: WP_008360004.1 replacing with: gerT\n",
      "Looking at: WP_071168069.1 replacing with: lipC\n",
      "Looking at: WP_083381142.1 replacing with: lipC\n",
      "Looking at: WP_071168649.1 replacing with: oxdD\n",
      "Looking at: WP_008360620.1 replacing with: oxdD\n",
      "Looking at: WP_008361546.1 replacing with: oxdD\n",
      "Looking at: WP_008361014.1 replacing with: safA\n",
      "Looking at: WP_008357345.1 replacing with: safA\n",
      "Looking at: WP_008359873.1 replacing with: safA\n",
      "Looking at: WP_008359989.1 replacing with: safA\n",
      "Looking at: WP_008359989.1 replacing with: safA\n",
      "Looking at: WP_008356730.1 replacing with: spoIVA\n",
      "Looking at: WP_071168843.1 replacing with: spoVID\n",
      "Looking at: WP_071168843.1 replacing with: spoVID\n",
      "Looking at: WP_008354550.1 replacing with: spoVM\n",
      "Looking at: WP_071168397.1 replacing with: spsB\n",
      "Looking at: WP_071168402.1 replacing with: spsI\n",
      "Looking at: WP_071168488.1 replacing with: spsI\n",
      "Looking at: WP_008360534.1 replacing with: spsI\n",
      "Looking at: WP_008355852.1 replacing with: spsI\n",
      "Looking at: WP_071169117.1 replacing with: sspO/cotK\n",
      "Looking at: WP_071168937.1 replacing with: tasA/cotN\n",
      "Looking at: WP_071168694.1 replacing with: tgl\n",
      "Looking at: WP_071168258.1 replacing with: yaaH\n",
      "Looking at: WP_008359873.1 replacing with: yaaH\n",
      "Looking at: WP_008357345.1 replacing with: yaaH\n",
      "Looking at: WP_071168258.1 replacing with: ydhD\n",
      "Looking at: WP_008361321.1 replacing with: yhaX\n",
      "Looking at: WP_008355235.1 replacing with: yhaX\n",
      "Looking at: WP_008357079.1 replacing with: yhaX\n",
      "Looking at: WP_008355652.1 replacing with: yhaX\n",
      "Looking at: WP_071167674.1 replacing with: yhaX\n",
      "Looking at: WP_071168455.1 replacing with: yhaX\n",
      "Looking at: WP_071168455.1 replacing with: yhaX\n",
      "Looking at: WP_008354793.1 replacing with: yhaX\n",
      "Looking at: WP_034739126.1 replacing with: yhbaqueg\n",
      "Looking at: WP_071167858.1 replacing with: yhbB\n",
      "Looking at: WP_008358712.1 replacing with: yhcN\n",
      "Looking at: WP_008354755.1 replacing with: yhcN\n",
      "Looking at: WP_071167854.1 replacing with: yhcQ\n",
      "Looking at: WP_008361316.1 replacing with: yheC\n",
      "Looking at: WP_008361313.1 replacing with: yheC\n",
      "Looking at: WP_008361313.1 replacing with: yheD\n",
      "Looking at: WP_008355136.1 replacing with: yhjR\n",
      "Looking at: WP_034738998.1 replacing with: yisY\n",
      "Looking at: WP_071168688.1 replacing with: yisY\n",
      "Looking at: WP_071168731.1 replacing with: yisY\n",
      "Looking at: WP_071167690.1 replacing with: yjqC\n",
      "Looking at: WP_008355274.1 replacing with: yjzB\n",
      "Looking at: WP_071167685.1 replacing with: yknT\n",
      "Looking at: WP_071169050.1 replacing with: ykvP\n",
      "Looking at: WP_008359873.1 replacing with: ykvP\n",
      "Looking at: WP_071168258.1 replacing with: ykvQ\n",
      "Looking at: WP_008359989.1 replacing with: ykzQ\n",
      "Looking at: WP_008359989.1 replacing with: ykzQ\n",
      "Looking at: WP_008359989.1 replacing with: ykzQ\n",
      "Looking at: WP_008359989.1 replacing with: ykzQ\n",
      "Looking at: WP_008357345.1 replacing with: ykzQ\n",
      "Looking at: WP_008354722.1 replacing with: ylbD\n",
      "Looking at: WP_071167760.1 replacing with: yncD\n",
      "Looking at: WP_071168044.1 replacing with: yncD\n",
      "Looking at: WP_008358101.1 replacing with: ypeP\n",
      "Looking at: WP_071169023.1 replacing with: yppG\n",
      "Looking at: WP_007496614.1 replacing with: yrbC\n",
      "Looking at: WP_071168844.1 replacing with: ysxE\n",
      "Looking at: WP_071168488.1 replacing with: ytdA\n",
      "Looking at: WP_008360534.1 replacing with: ytdA\n",
      "Looking at: WP_071168402.1 replacing with: ytdA\n",
      "Looking at: WP_008355852.1 replacing with: ytdA\n",
      "Looking at: WP_008356662.1 replacing with: yusN\n",
      "Looking at: WP_071168655.1 replacing with: yutH\n",
      "Looking at: WP_034738645.1 replacing with: yuzC\n",
      "Looking at: WP_008355538.1 replacing with: ywqH\n",
      "Looking at: WP_071168360.1 replacing with: ywqH\n",
      "Looking at: WP_034740044.1 replacing with: ywqH\n",
      "DOne!\n"
     ]
    }
   ],
   "source": [
    "# Steps to extract genes from pBLAST results for each genome/species\n",
    "\n",
    "# Step 3: Change headers id from extracted genes (optional but helpful in future steps)\n",
    "\n",
    "from Bio import SeqIO\n",
    "\n",
    "#genes list\n",
    "#the list can be coppied from the .csv file (second column) generated in step 1 (this is why we saved the \"record.query\")\n",
    "genes_list = [\"cgeA\",\n",
    "\"cgeA\",\n",
    "\"cgeB\",\n",
    "\"cgeC\",\n",
    "\"cgeD\",\n",
    "\"cgeD\",\n",
    "\"cgeD\",\n",
    "\"cgeD\",\n",
    "\"cotA\",\n",
    "\"cotD\",\n",
    "\"cotE\",\n",
    "\"cotF\",\n",
    "\"cotH\",\n",
    "\"cotI\",\n",
    "\"cotJC\",\n",
    "\"cotM\",\n",
    "\"cotO\",\n",
    "\"cotP\",\n",
    "\"cotS\",\n",
    "\"cotSA\",\n",
    "\"cotSA\",\n",
    "\"cotSA\",\n",
    "\"cotSA\",\n",
    "\"cotV\",\n",
    "\"cotV\",\n",
    "\"cotW\",\n",
    "\"cotX\",\n",
    "\"cotY\",\n",
    "\"cotY\",\n",
    "\"cotZ\",\n",
    "\"cotZ\",\n",
    "\"coxA/yrbB\",\n",
    "\"cwIJ\",\n",
    "\"cwIJ\",\n",
    "\"cwIJ\",\n",
    "\"gerPA\",\n",
    "\"gerPA\",\n",
    "\"gerPA\",\n",
    "\"gerPA\",\n",
    "\"gerPB\",\n",
    "\"gerPC\",\n",
    "\"gerPD\",\n",
    "\"gerPE\",\n",
    "\"gerPF\",\n",
    "\"gerPF\",\n",
    "\"gerPF\",\n",
    "\"gerPF\",\n",
    "\"gerQ\",\n",
    "\"gerR/ylbO\",\n",
    "\"gerR/ylbO\",\n",
    "\"gerT\",\n",
    "\"lipC\",\n",
    "\"lipC\",\n",
    "\"oxdD\",\n",
    "\"oxdD\",\n",
    "\"oxdD\",\n",
    "\"safA\",\n",
    "\"safA\",\n",
    "\"safA\",\n",
    "\"safA\",\n",
    "\"safA\",\n",
    "\"spoIVA\",\n",
    "\"spoVID\",\n",
    "\"spoVID\",\n",
    "\"spoVM\",\n",
    "\"spsB\",\n",
    "\"spsI\",\n",
    "\"spsI\",\n",
    "\"spsI\",\n",
    "\"spsI\",\n",
    "\"sspO/cotK\",\n",
    "\"tasA/cotN\",\n",
    "\"tgl\",\n",
    "\"yaaH\",\n",
    "\"yaaH\",\n",
    "\"yaaH\",\n",
    "\"ydhD\",\n",
    "\"yhaX\",\n",
    "\"yhaX\",\n",
    "\"yhaX\",\n",
    "\"yhaX\",\n",
    "\"yhaX\",\n",
    "\"yhaX\",\n",
    "\"yhaX\",\n",
    "\"yhaX\",\n",
    "\"yhbaqueg\",\n",
    "\"yhbB\",\n",
    "\"yhcN\",\n",
    "\"yhcN\",\n",
    "\"yhcQ\",\n",
    "\"yheC\",\n",
    "\"yheC\",\n",
    "\"yheD\",\n",
    "\"yhjR\",\n",
    "\"yisY\",\n",
    "\"yisY\",\n",
    "\"yisY\",\n",
    "\"yjqC\",\n",
    "\"yjzB\",\n",
    "\"yknT\",\n",
    "\"ykvP\",\n",
    "\"ykvP\",\n",
    "\"ykvQ\",\n",
    "\"ykzQ\",\n",
    "\"ykzQ\",\n",
    "\"ykzQ\",\n",
    "\"ykzQ\",\n",
    "\"ykzQ\",\n",
    "\"ylbD\",\n",
    "\"yncD\",\n",
    "\"yncD\",\n",
    "\"ypeP\",\n",
    "\"yppG\",\n",
    "\"yrbC\",\n",
    "\"ysxE\",\n",
    "\"ytdA\",\n",
    "\"ytdA\",\n",
    "\"ytdA\",\n",
    "\"ytdA\",\n",
    "\"yusN\",\n",
    "\"yutH\",\n",
    "\"yuzC\",\n",
    "\"ywqH\",\n",
    "\"ywqH\",\n",
    "\"ywqH\"]\n",
    "    \n",
    "#open and create output files    \n",
    "with open(\"Bacillus_xiamenensis_VV3_genes.fas\") as original, open(\"Bacillus_xiamenensis_VV3_genes_rename.fas\", \"w\") as corrected: \n",
    "    records = SeqIO.parse(\"Bacillus_xiamenensis_VV3_genes.fas\", \"fasta\") #parse input file\n",
    "    i = 0\n",
    "    for record in records: #iterate over each line  of input file (each line start with \">\", i.e. a record)\n",
    "        print(\"Looking at: \" + record.id, \"replacing with: \" + genes_list[i])\n",
    "        record.description = genes_list[i]\n",
    "        record.id = genes_list[i] \n",
    "        record.name = genes_list[i]\n",
    "        #print(record.description)\n",
    "        i = i+1\n",
    "        SeqIO.write(record, corrected, 'fasta')\n",
    "           \n",
    "print(\"DOne!\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Looking at file: Bacillus_altitudinis_SGAir0031_genes_rename.fas\n",
      "Extracting : cotE\n",
      "Replacing header: cotE by B_altitudinis\n",
      "DOne here!\n",
      "Looking at file: Bacillus_pumilus_SH-B9_genes_rename.fas\n",
      "Extracting : cotE\n",
      "Replacing header: cotE by B_pumilus\n",
      "DOne here!\n",
      "Looking at file: Bacillus_safensis_KCTC12796BP_genes_rename.fas\n",
      "Extracting : cotE\n",
      "Replacing header: cotE by B_safensis\n",
      "DOne here!\n",
      "Looking at file: Bacillus_xiamenensis_VV3_genes_rename.fas\n",
      "Extracting : cotE\n",
      "Replacing header: cotE by B_xiamenensis\n",
      "DOne here!\n",
      "FInally DOne!!\n"
     ]
    }
   ],
   "source": [
    "# Steps to creat a database of the same gene from multiple genomes/species (used for \"Selection methods\")\n",
    "\n",
    "# Step 1: Extract the same gene from multiple files (i.e. species)\n",
    "\n",
    "from Bio import SeqIO\n",
    "\n",
    "#list of fasta files\n",
    "list_input_fasta=[\"Bacillus_altitudinis_SGAir0031_genes_rename.fas\",\n",
    "                 \"Bacillus_pumilus_SH-B9_genes_rename.fas\",\n",
    "                 \"Bacillus_safensis_KCTC12796BP_genes_rename.fas\",\n",
    "                 \"Bacillus_xiamenensis_VV3_genes_rename.fas\"] \n",
    "\n",
    "output_file = open(\"cotE.fas\", \"a+\") #append to a file, \"+\" if  file does not exist, then it will be created\n",
    "\n",
    "for file in list_input_fasta: #iterate over fasta files\n",
    "    handle = SeqIO.parse(file, \"fasta\") #open fasta file\n",
    "    print(\"Looking at file: \" + file)\n",
    "    for record in handle: #iterate in the open fasta file\n",
    "        if record.id == \"cotE\": #gene name\n",
    "            name = file.split(\"_\")\n",
    "            id = \"B_\" + name[1]\n",
    "            print(\"Extracting : \" + record.id)\n",
    "            print(\"Replacing header: \" + record.id + \" by \" + id)\n",
    "            #print(record.seq)\n",
    "            new_seq = record.seq #extract seq\n",
    "            output_file.write(\">%s\\n%s\\n\" % (id, new_seq)) #write id + seq\n",
    "    print(\"DOne here!\")\n",
    "    handle.close()\n",
    "output_file.close()\n",
    "print(\"FInally DOne!!\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Looking at: B_safensis\n",
      "Replacing with new seq:  ATGTCTGAATACAGAGAAATTATTACAAAAGCGGTGGTTGCAAAAGGGAGGAAATTCACCCAAAGCACACACACCATTTCTCCATCGCAAAAACCAACCAGCATCCTAGGTGGTTGGATCATTAATCATAAGTATGATGCTGAAAAGGTCGGTAAAACCGTTGAGATCGAAGGAACATTTGATATTAACGTATGGTACTCTTACGCCGACAATACGAAAACAGAAGTCGTCACAGAGCGTGTGAAGTATGTAGACATCATTAAACTGAGATATAAAGATAAAAATTTCCTTGATGATGAGCACGAAGTAATTGCGAAAGTATTGCAGCAGCCCAATTGCTTAGAAGTGACCATTTCTCCAAATGGCAATAAAATTATCGTTCAGGCAGAGCGTGAATTCATTGCTGAGGTAGTCGGTGAAACAAAAATTGTCGTAGAGGTCGATTCAAGCTGGAAAGAAAAAGATGATCACGAGTGGGAAGAAGAAGTGGATGAGGAGTTAGAGGATATTCATCCAGAATTCCTAGCAGGTGAGACAGAAGAG---\n",
      "\n",
      "Looking at: B_altitudinis\n",
      "Replacing with new seq:  ATGTCTGAATACAGAGAAATTATTACAAAAGCGGTGGTTGCAAAAGGGAGGAAATTTACCCAAAGCACACACACCATTTCTCCATCGCAAAAACCAACCAGCATCCTAGGCGGTTGGATCATTAATCATAAGTATGATGCTGAAAAGGTCGGTAAAACAGTTGAGATTGAAGGTACATTTGATATTAACGTGTGGTATTCTTACGCAGACAACACGAAAACCGAAGTGGTCACAGAACGTGTAAAATATGTAGATATCATCAAACTGAGATATAAAGATAAAAACTTTCTTGATGATGAGCACGAAGTGATTGCGAAAGTCTTACAGCAGCCCAATTGCTTAGAAGTCACCATTTCTCCAAATGGCAATAAAATTGTCGTTCAAGCAGAGCGTGAATTCATTGCTGAAGTAGTAGGGGAAACCAAAATTGTTGTAGAAGTCAATTCAAGCTGGAAAGAAAAAGATGATCACGAATGGGAAGAAGAAGTGGATGAGGAATTAGAAGATATTCATCCAGAATTCCTCGCGGGTGATCCAGAAGAG---\n",
      "\n",
      "Looking at: B_xiamenensis\n",
      "Replacing with new seq:  ATGTCTGAATACAGAGAGATTATTACAAAAGCGGTGGTTGCAAAAGGGAGGAAATTTACCCAAAGCACACACACCATTTCTCCATCGCAAAAACCAACCAGCATCCTAGGCGGTTGGATCATTAATCATAAGTATGATGCTGAAAAGGTCGGTAAAACAGTTGAAATTGAAGGAACATTTGATATTAACGTGTGGTATTCTTACGCAGACAACACGAAAACCGAAGTGGTCACAGAACGTGTGAAATATGTAGATATCATCAAACTGAGATATAAAGATAAAAACTTTCTCGATGATGAGCACGAAGTGATTGCGAAAGTCTTACAGCAGCCCAATTGCTTAGAAGTCACCATCTCTCCAAATGGCAATAAAATTGTCGTTCAGGCAGAGCGTGAATTCATTGCAGAAGTAGTAGGAGAAACAAAAATTGTGGTAGAGGTCAATTCAAGCTGGAAAGAAAAAGATGATCATGAGTGGGAAGAAGAAGTGGATGAGGAGTTAGAAGACATTCATCCAGAATTCCTAGCAGGTGATCCTGAAGAG---\n",
      "\n",
      "Looking at: B_pumilus\n",
      "Replacing with new seq:  ATGTCTGAATACAGAGAGATTATTACAAAAGCGGTGGTTGCAAAAGGGAGGAAATTCACCCAAAGCACACACACCATTTCTCCATCGCAAAAACCAACCAGCATCCTAGGTGGTTGGATCATTAATCATAAGTATGATGCTGAAAAGGTCGGTAAAACTGTTGAGATTGAAGGAACCTTTGATATTAACGTATGGTATTCTTACGCCGATAACACGAAAACAGAAGTCGTCACAGAACGGGTGAAATATGTAGACATCATCAAACTGAGATATAAAGATAAAAATTTCCTTGATGATGAGCATGAAGTGATTGCAAAAGTGCTCCAGCAGCCGAATTGCTTAGAAGTGACCATTTCTCCAAACGGCAATAAAATTATCGTTCAAGCAGAGCGGGAATTCATTGCTGAAGTGGTAGGGGAAACAAAAATTGTTGTAGAGGTCAATTCAAGCTGGAAAGAAAAAGATGATCATGAGTGGGAAGAAGAGGTGGATGAGGAGTTAGAGGATATTCATCCAGAATTCCTAGCGGGTGATCCAGAAGAG---\n",
      "\n",
      "DOne!\n"
     ]
    }
   ],
   "source": [
    "# Steps to creat a database of the same gene from multiple genomes/species (used for \"Selection methods\")\n",
    "\n",
    "#Step 2: Delete stop codons of a gene database v1\n",
    "\n",
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "\n",
    "codon_stop = [\"TAG\", \"TGA\", \"TAA\"]\n",
    "\n",
    "input_file = \"cotE_aligned.fas\" #name of the file (the file must be aligned, so all sequences have the same length!)\n",
    "\n",
    "handle = SeqIO.parse(input_file, \"fasta\")\n",
    "for record in handle: #open and iterate your file\n",
    "    print(\"Looking at: \" + record.id)\n",
    "    #print(record.seq )\n",
    "    tempRecordSeq = list(record.seq) #make the sequence a list (each letter is an element of the list)\n",
    "    for index in range(0, len(record.seq), 3): #iterate each time over 3 letters (codon) of the sequence\n",
    "        codon = record.seq[index:index+3] #recall that [i:i+3] is not inclusive, so it wil take from i to i+2\n",
    "        if codon in codon_stop: #check if the codon is a stop codon\n",
    "            tempRecordSeq[index:index+3] = '-','-','-' #replace each letter of the stop codon with \"-\" to not disturb the length of the alignment\n",
    "    record.seq = Seq(\"\".join(tempRecordSeq)) #define a new sequence by joining the elements of the list\n",
    "    print(\"Replacing with new seq: \", record.seq + \"\\n\")\n",
    "    name = input_file.split(\".\")\n",
    "    output_file = open(name[0] + \"_nostop.fas\", \"a+\") #output file for the alignment\n",
    "    output_file.write(\">%s\\n%s\\n\" % (record.id, record.seq)) #write id + seq\n",
    "handle.close()\n",
    "output_file.close()\n",
    "print(\"DOne!\")\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Looking at: cgeA_aligned.fas\n",
      "Looking at: B_subtilis\n",
      "Replacing with new seq:  ATGAGCTCTGAAAATGCACAGTTAAAAAAGGATTTAATTAAAGCCGTTTTAAGCCCTTTATTTCCAACTGCA------ACA---------GAAGGAGGAGAAAATATGGATAGTAATCTTAAAGCCTTGCTTGATGCTGCCATCGATCAAAAAGTAGAT---------GAAAGTGAAACGGTTACGGCAGAATCTATTTTAGACCCT------TCTCTTCCGGCAAGATGGATTTTTGCCAGAATTACGCCAGGCACTACAATCAGCATTGTGACTGATTCAGGTGACATGATCGGACCGGTTGTTTTCGTTGCTTTCGATCAGGTTCACGGGATCGTATTTGTAACACAGGAAAGCTCCGTCACTCCGGCAGGCCAAGCTACAACATTAATTGATGTAGACAAAGTAGAAAGCGTTACGTTCTTTTCA---\n",
      "\n",
      "Looking at: B_amyloliquefaciens\n",
      "Replacing with new seq:  ATGAGCGAAGATCGTTCAGAGCTCAAAAAGAAGCTATTGAAAACTCTTATTGCCCCTATATTTGACAACAGCGCCTTATCTAGCGGCACAGATGGAGGAGATAACGTGGAAAACAAGGTAAAAGATTTATTAGAATCTGCAATTGATGCAAAAGTGGATGAAGCTGCTGCGAAAGACAGTCTCAGCACTCAATCTGTTCTGGGAAGCGGTATTTCACTAGCCGCGAGATGGATTTTTGCAAGAGTTCAGCCCGGCACTATCATCAGCATTGTGATGGATTCTGGAGACATGATCGGCCCGGTTCAATTTGTTTTCTTTGACGAAGTTCACGGCATCGTTTATGTGACGCAGGAAAATTCCGTTACACCGGCAGGCTCTTCTAAAACATTGCTTGACGTTGACAAAGTTGAAAGTGTCACGTTCACTTCG---\n",
      "\n",
      "Looking at: B_atrophaeus\n",
      "Replacing with new seq:  ATGAGTAATGATTTTCTGCAGTTGAAAAAGGATCTCATTAAAAGTTTATTGATCCCCTTTTTTAATAATGGA------ACGACAACTGAAAGTGGAGGAGATAACGTGGAAAACAAAATTAAAGAATTATTAGAAGCAGCGGTAGACAATAAAGTAGATGCAGCTGCTGAAGCAGAAGCCGCAAGCGATGTATCAGCCTTGGATACG------TCCTTACCGGCCAGATGGATTTTTGCTAGAATTACACCGGGAACTGTAATCAGCATTGTGATGGATTCAGGAGACATGATCGGACCTGTTCAATTTGTTGCATTTGACCAAATCCACGGCATTGTATATGTCACGCAAGAAGCTTCCGTCACTCCGACTGGTTCAGCCACTACATTGCTTGATGTAGATAAAGTAGAAAGCGTTACGTTCACTACG---\n",
      "\n",
      "Looking at: B_gibsonii\n",
      "Replacing with new seq:  ATGAGCTCTGAAAATGCACAGTTAAAAAAGGATTTAATTAAAGCCGTTTTAAGCCCTTTATTTCCAACTGCA------ACA---------GAAGGAGGAGAAAATATGGATAGTAATCTTAAAGCCTTGCTTGATGCTGCCATCGATCAAAAAGTAGAT---------GAAAGTGAAACGGTTACGGCAGAATCTCTTTTAGACCCT------TCTCTTCCGGCAAGATGGATTTTTGCCAGAATTACGCCAGGCACTACAATCAGCATTGTGACTGATTCAGGTGACATGATCGGACCGGTTGTTTTCGTTGCTTTCGATCAGGTTCACGGGATCGTATTTGTAACACAGGAAAGCTCCGTCACTCCGGCAGGCCAAGCTACAACATTAATTGATGTAGACAAAGTAGAAAGCGTTACGTTCTTTTCA---\n",
      "\n",
      "Looking at: B_halotolerans\n",
      "Replacing with new seq:  TTGAGCAGTGAAAATGCGCAGTTAAAAAAGGATATTATTAAAAGCCTTTTAAGCCCTTTATTTACAAATGCA------ACA---------GAAGGAGGAGAAAACGTGAATAATAACCTTAAAGAATTACTTGAAGCTGCTGTTGATAAAAAAGTTGAC---------GAAAGCGAAACGGTTACTGCTGAATCTGTTTTAGATCCT------TCCCTTCCTGCAAGATGGATTTTTGCCAGAATTACACCTGGCACTACCATCAGCATTGTGACAGATTCCGGTGACATGATCGGTCCGGTTGTCTTTGTTGCTTTTGATCAAGTTCACGGCATCGTATTTGTCACACAGGAAAGCTCTGTAACTCCGCCAGGTCAAGCGACAACATTGATTGATGTAGATAAAGTAGAAAGCGTTACATTCTTTTCA---\n",
      "\n",
      "Looking at: B_intestinalis\n",
      "Replacing with new seq:  ATGAGCTCTGAAAATGCACAGTTAAAAAAGGATTTGATTAAAACCCTTTTAAGCCCTTTATTC------TCA------ACA---------GAAGGAGGAGTAAACGTGGATAATAATCTTAAAGACATGCTTGAAGCTGCCATCGATCAAAAAGTAGAT---------GAAAGTGAAATGATTACGGCTGAATCTACT---------------TCCCTTCCGGCAAGATGGATTTTTGCCAGAGTTACGCCAGGCACGACAATCAGCATTGTGACTGATTCCGGTGACATGATCGGGCCGGTTGTGTTCGTTGCTTTCGATCAAGTTCATGGTATCGTATTTGTAACACAGGAAAGCTCCGTCACTCCGGCAGGTCAAGCTACAACATTAATTGATGTAGACAAAGTAGAAAGCGTTACGTTCTTTTCA---\n",
      "\n",
      "Looking at: B_siamensis\n",
      "Replacing with new seq:  ATGAGCGAGGATCGTTTAGAGCTCAAAAAGAAGCTATTGAAAACTCTTTTGAACCCTATATTTGACAACAGCGCTTTATCTAGCGGCACAGATGGAGGAGATAACGTGGAAAACAATATAAAAGATTTATTAGAATCTGCAATTGATGCAAAAGTGGATGAAGCAGCTGCGCAAGACAGTCTCAGCGCTCAATCTGTTTTAGGAAGCGGCATTTCATTAGCGGCGAGATGGATTTTTGCGAGAGTTCAGCCCGGCACTGTTATCAGCATTGTGATGGATTCAGGAGACATGATCGGTCCGGTACAATTTGTTTTCTTTGACGAAGTTCACGGAATTGTTTATGTTACGCAGGAAAATTCCGTTACACCGGCAGGCTCTGCTACAACATTGCTTGACGTTGACAAAGTCGAAAGTGTCACCTTCACTTCG---\n",
      "\n",
      "Looking at: B_vallismortis\n",
      "Replacing with new seq:  ATGAGCGAGGATCGTTTGGAGCTTAAAAAGAACTTATTGAAAACTCTTTTAGCCCCTGTATTTGACAACAGCGCTTTATCTGGCGGCACAGATGGAGGAGATAACGTGAAAAACAGCGTAAAAGAATTATTGGAATCGGCAATTGATGCAAAAGTAGATGAAGCAGCGGCGAAAGACAGTCTCAGCGCACAATCTGTATTGGGAAGCGGTATTTCATTAGCGGCGAGATGGATTTTTGCAAGAATCCAGCCCGGCACTGTCATCAGCATTGTGATGGATTCAGGAGACATGATCGGCCCGGTTCAATTTGTTGCCTTTGACGAAATTCACGGAATCGTTTTTGTTACGCAGGAAAATTCCGTTACACCGGCAGGCTCTGCTACAACATTGCTTGACGTCGACAAAGTTGAAAGTGTCACGTTCACTTCG---\n",
      "\n",
      "Looking at: B_velezensis\n",
      "Replacing with new seq:  ATGAGCGAAGATCGTTTGGAGCTTAAAAAGAACTTATTGAAAACTCTTTTAGCCCCTGTATTTGACAACAGCGCTTTATCTGGCGGTACAGATGGAGGAGATAACGTGAAAAACAGCGTAAAAGAATTATTGGAAGCGGCAATTGATGCAAAAGTAGATGAAGCAGCGGTGAAAGACAGTCTCAGCGCACAATCTGTATTGGGAAGCGGTATTTCATTAGCGGCGAGATGGATTTTTGCAAGAATTCAGCCCGGCACTGTCATCAGCATTGTGATGGATTCAGGAGACATGATCGGCCCGGTTCAATTTGTTGCCTTTGACGAAATTCACGGAATCGTTTTTGTTACGCAGGAAAATTCCGTTACACCGGCAGGCTCTGCTACAACATTGCTTGACGTCGACAAAGTTGAAAGTGTCACGTTCACTTCG---\n",
      "\n",
      "DOne here!\n",
      "\n",
      "Looking at: cgeB_aligned.fas\n",
      "Looking at: B_subtilis\n",
      "Replacing with new seq:  ATGAAAGTCTTGTATATCCAGTCGGGATACGGAGGAATATACAGTTATTTTGATCGCTGGGCTGAAGAATGTTTTCAGAACACTCATACAGAGTATATGATTGCAGATAAACCTGAAGCAGAATCTTTGATGAAGATCGAAGCGTTTCAGCCGGATTTCACACTTATGATGGTTGGAGACCGCGTTCCTCACGACTGGCTGACCTGGTTAAAGGGTAAGGATATCCCCGTGTATGTTTGGCTGACCGAAGACCCATTTTATATGGATATCAGCCTTCAGGTAATTAAGCTTGCTGACGCCATATTAACGATAGAACAAAATGCAGCTCTCTACTATCAAGAGCTCGGCTACCAGAACGTCTATTATGTTCCGATACCAGTTAACCATCGGCTGTTTAAAAAAATGGGCACCGAACATTCCTATCATTCAAATCTGTTAATCATCGGCTATCCTTACCCTAATCGGGTTCAGCTTATGAAAGAAGCGGTCCATCTGCCATTCACGGTTCGGGTAATCGGCAAGGAATGGGGAAAATATCTGCCTAAAAAAGTGCTCAAACAGCCGCATATTGATGTCGTCAGCACATGGGTTCCACCCGAACAGGCCGTTCATTATTATAATGGGGCAGACATCGTAATTAATGCGCATCGTCCCTATCATTTTGCTTTCAACCAAAATACTATGCGCATCAAAAATGCCAGTTTTAATAACCGAACGTTTGATATCGCTGCATGCGAAGCATTTCAATTAACAGATTTACCCGCGGCGCATCCTTTTTCATCCATTATTTCTTATCACGGCATGAACGACTTTAAGGAAAAAGCCGCTTTCTATATCAACCACCCTGAAGAACGGCAAAAAGCTGCAGCAGCCAATTATAAAGAGACCGTACCCGCATTTACGTTTGATGAGCTTCCTGCAAAACTAAAAGCGATTCATCTGGCGCTTTCT------\n",
      "\n",
      "Looking at: B_amyloliquefaciens\n",
      "Replacing with new seq:  ATGAAAGTTTTATATATCCGTTCAGGATATAAAGGCATTTACACGTATTTTGACCGCTGGATCGAGGAAGGCTTTTTACAGTCTTCCGTCCGCTTTTTTTCTGTTTCTGAGCTTCAGGAAGATTCAATCCGCAGTATACGGGCATTCAAACCTGACATCAGTCTCATGATGGCGGGCGACCGTATTCCGTCTGAATGGCTTCAATGGCTGAGGACAGAAGAGATTCCCGTCTATCTATGGTTGACGGAGGATCCTTTTTATTTCGATGTGAGCAAGACAATCGCCCCGCTCGCCGATGCCGTTTTGACCATAGAACAAAACGCTCTTGATGCTTATCGGCAAATGGGCTTACAGCACGTATATTATGTCCCGATTCCCGTTAATCAGCGGCTTTTTACAAAACAAGCGGCTGACGCTTCTCTTCACACAAACCTGCTTCTGATCGGATACCCGTACCCGAATCGGGTTGAACTGGTGAAAGCAGCGGCGAAGCTTCCGTATACGCTTCGGATCATCGGGAAGGGATGGCGCAGACATCTGCCTAAAAAAATCGCCAGACAGCAAAATGTGCTCATAATGGATGAATGGATTGAACCCGAACATGCCGCATTGCATTACAACAGCGCAGATGTGGTTTTGAACCCGCACCGTTCTTATCGGTTTGCGCTCAATCAAAATAGGAGCGGCATCCGAAATATCAGTTTAAATACGAGAAGCTTTGACATTGCGGCTTGTCAGGCATTTCAGCTGACAGACCTTTCTCCCGCGCCCCCTTTTTCATCCTTTGTTTCTTACAGCGGGCTTGCGGATTTTGCGGAAAAGATCCATTTTTATATGACAAACCCACGTGAACGTGAACAAATCGCGGCGCTTAATTATCAAGAGACAGTGCCTGCTTACACGTACGACCGGATTCCGGCTGTGCTCACGGCTATCCGCTCCAAAACTTTCTCA---\n",
      "\n",
      "Looking at: B_atrophaeus\n",
      "Replacing with new seq:  ATGAACGTATTGTATATCCGTTCGGGATATAAGGGGATATACAGCTATTTTGACCGCTGGATTGAAGAAGGTTTTCTTCATTCTCCCATGAGGCTCCTTTCTTCACTTGATCTTGATGCGCAATCAGCCCGCAGCATCAAATCATTCCAGCCGGATTTTGCTGTCATTATGGTCGGAGATAGAGTCCCTTCTGAATGGCTGACCTGGTTAAAGATTGAGGGTATTCCCGTGTATGTCTGGCTTACAGAAGACCCTTTTTATACAGATATCAGCCTGCAAGTCATACCGCATGCAGATGTGATATTGACCATTGACCAAAATTCAGCCGATGTTTACCGGGAATTAGGCTGCCCCCACGTTTATTACATGCCCATTCCCACAAACCAGCGGCTTTTCAAAAAACTCAATACGGAAACTTCCTGCCATTCTAACCTGCTGCTTATCGGCTATCCTTATCCTAACCGGGTG------------------------------------------------------------------------------------------------------GATATTATCGACCAATGGATCAGCCCTGAAAAAGCAGTTCATTATTATAACAGCGCGGACATTGTCATTAATCCGCACAGGCCGTACCATTTTGCGTTCAATAAAAACCAGAAGTCCATTCAAAATATCAGCTTAAATAACCGAAGCTTTGACATTGCAGCTTGTGAAGCGTTTCAGCTCGTAGATCTTCCTGCGGCAAGTCCCTTTTCATCCTTTGTTTCTTATCACGGAATAAACGATTTTAAGCGTAAAGCAGCATATTATCTGGAGCACCCTGAAGAACGAAAAAAAGCCGCGGCACTGAATTATAAAGAGACAATTCCAGCCAATACGTTTGATGAACTTCCGGCTAAACTCTCACACATTCATTTAAATTGT---------\n",
      "\n",
      "Looking at: B_gibsonii\n",
      "Replacing with new seq:  ATGAAAGTCTTGTATATCCAGTCGGGATACGGAGGAATATACAGTTATTTTGATCGCTGGGCTGAAGAATGTTTTCAGAACACTCATACAGAGTATATGATTGCAGATAAACCTGAAGCAGAATCTTTGATGAAGATCGAAGCGTTTCAGCCGGATTTCACACTTATGATGGTTGGAGACCGCGTTCCTCACGACTGGCTGACCTGGTTAAAGGGTAAGGATATCCCCGTGTATGTTTGGCTGACCGAAGACCCATTTTATATGGATATCAGCCTTCAGGTAATTAAGCTTGCTGACGCCATATTAACGATAGAACAAAATGCAGCTCTCTACTATCAAGAGCTCGGCTACCAGAACGTCTATTATGTTCCGATACCAGTTAACCATCGGCTGTTTAAAAAAATGGGCACTGAACATTCCTATCATTCAAATCTGTTAATCATCGGCTATCCTTACCCTAATCGGGTTCAGCTTATGAAAGAAGCGGTCCATCTGCCATTCACGGTTCGGGTAATCGGCAAGGAATGGGGAAAATATCTGCCTAAAAAAGTGCTCAAACAGCCGCATATTGATGTCGTCAGCACATGGGTTCCACCCGAACAGGCCGTTCATTATTATAATGGGGCAGACATCGTAATTAATGCGCATCGTCCCTATCATTTTGCTTTCAACCAAAATACTATGCGCATCAAAAATACCAGTTTTAATAACCGAACGTTTGATATCGCTGCATGCGAAGCATTTCAATTAACAGATTTACCCGCGGCACATCCTTTTTCATCCTTTATTTCTTATCACGGCATGAACGACTTTAAGGAAAAAGCCGCTTTCTATATCTACCACCCTGAAGAACGGCAAAAAGCTGCAGCAGCCAATTATAAAGAGACCGTACCCGCATTTACGTTTGATGAGCTTCCTGCAAAACTAAAAGCGATTCATCTGGCGCTTTCT------\n",
      "\n",
      "Looking at: B_halotolerans\n",
      "Replacing with new seq:  ATGAAAGTTTTGTATATCCAGTCGGGATATGGAGGAATATACAGTTATTTTGACCGCTGGGCTGAAGAATGCTTTCGGCACGCTCAGACAGAATACATGGCTGCAGATACACCAGACACCGAATCGCTGTCACGCATTGAAACGTTTCAGCCGGATTTCGTACTCATGATGGTTGGAGTTCACGTTCCTCACGACTGGCTGACCTGGTTAAAGGGTAAGGGTATTCCCGTATATGTTTGGCTGACAGAAGACCCATTTTATATTGATATCAGTCTTCAGGTGATAAAGCACGCTGACGCAGTTTTAACAATTGACCAAAATGCTGCTCTCTATTATCAGGAGCTTGGTTATCAGCATATCTATTATGTTCCGATTCCTGTTAATCATCGGCTGTTTAAAAAAATGGACACCGAACATTCCTTTCATTCACAATTGCTTATCATCGGCTATCCTTACCCTAATCGGGTTCAGCTTGTGAAAGAAGCGGCATATCTCCCGTACACCGTTCGGGTCATCGGAAAGGAATGGAGAAAATACCTGCCTAAAAAAGTACTCAAACAGCCGCATATAGATGTAGTCAGCACATGGGTTCCGCCTGAACAGGCCGTTCATTATTATAACGGGGCAGACATCGTGATCAATCCGCATCGGCCGTATCATTTTATGTTCAACAAAAATAAAAGCCGCATTAAAAATACAAGCTTTAATAACCGCACCTTTGATATCGCTGCATGCGAAGCATTTCAACTAGTGGATCTCCCTGCGGCACATCCTTTTTCATCCTTTATTTCTTATCACAGCATCAACGACTTTAAGGAAAAAGCCGCTTTCTATATCAATCATCCTGAGGAACGAAACAAAGCTGCGGCGTTAAATTATAAAGAGACCGTTCCTGCTCATACATTTGATGAGCTTCCCGCAAAACTGAAAGCCATTCATCTGGCGCTTTCT------\n",
      "\n",
      "Looking at: B_intestinalis\n",
      "Replacing with new seq:  ATGAAAGTCTTGTATATCCAGTCGGGATACGGAGGAATTTACAGTTATTTTGATAGCTGGGTTGAAGAATGTTTTCGGATCACTCATACAGAGTATATGTCTGCAGATAAACTGGAAGCAGCATCTCTGGTGAAGATCGAAGCGTTTCAGCCGGACTTCGCACTTATGATGGTTGGAGACCGCGTTCCTCACGACTGGCTGACCTGGTTAAAGGGTAAGGGTATTCCCGTGTATGTTTGGCTGACCGAAGACCCATTTTATATTGATATCAGTCTTCAGGTGATTAAGCACGCTGACGCCATATTAACGATTGAACAAAATGCCGCTCTCTATTATCGGGAGCTTGGCTTCCAGCACGTCTATTATGTTCCGATTCCAGTTAATCATCGGCTGTTTAAAAAAATGGACACCGAACATTCTTTTCATTCAAATCTGCTTATCATCGGCTATCCTTACCCTAATCGGGTTCAGCTTATGAAAGAAGCGGTACACCTGCCTTACACGGTTCGAGTCATCGGAACGGAATGGAAAAAATATCTGCCTAAAAAAGTGCTCAAACAGCCGCATATCGATGTCGTCAGCACATGGATTCCACCCGAACAGGCCGTTCATTATTATAACGGGGCAGACATTGTCATCAATGCGCATCGGCCTTATCATTTTGCTTTCAACAAAAATAATATTCGCATCAAAAATGCCAGCTTTAATAACCGAACGTTTGATATCGCTGCATGCGAAGCATTTCAATTAACAGATTTACCCGCGGCCCATCCTTTTTCATCCTTTATTTCTTATCACAGCATTAACGACTTTAAGGAAAAAGCTGCTTTCTATATCAACCATCCTGAGGAACGGCATAAAGCAGCTGCATCCAATTACAAAGAGACCGTACCCGCAAATACGTATGATGAGCTTCCTGAAAAACTAAAGGCGATTCATCTGGCGCTTTCT------\n",
      "\n",
      "Looking at: B_siamensis\n",
      "Replacing with new seq:  ATGAACGTTCTATATATCCGTTCAGGATATAAAGGCATTTACACGTATTTTGACCGCTGGATCGAGGAAGGTTTTTTACAGTCTTCCGTCCGCTTTTTTTCTGTTTCTGATCTTCAGGAAGATACAATCAGCAGCATACGGGAATTCAAACCTGACATCAGTCTGATGATGGCGGGCGACCGTATTCCGTCAGAATGGCTTCAATGGCTGAGAACAGAGCAGATTCCCGTCTATCTATGGCTGACGGAGGACCCTTTTTATTTTGATGTGAGCGCAAAAATCGCCCCGCTCGCCGATGCCGTTTTGACCATAGAACAAAACGCTCTTGATGCTTACCGGCAAATGGGCTTACAGCACGTATATTATGTCCCGATTCCCGTTAATCAGCGGATTTTTACGAAACAAGCGCCTGACGCTTCTCTTCACACCAATCTGCTTCTGGTCGGATATCCGTATCCGAATCGTGTTGAACTCGTGAAAGCAGCGGCGAAGCTTCCGTATACGCTTCGTATCATCGGGAAGGGATGGCGCAGACATCTGCCTAAAACAATCGCCAGACAGAAAAATGTGCTCATAATGGATGAATGGATTGAACCTGAACATGCCGCATTGTATTACAACAGCGCAGACGTGGTCTTAAACCCGCACCGTTCTTACCGGTTTGCGCTCAATCAGAATGGACGCGGCATCCAAAATATCAGTTTAAATACGAGAAGCTTTGACATTGCGGCCTGTCAGACATTTCAGCTGACAGATCTATCTCCCGCGCCCCCTTTTTCATCCTTTGTTTCTTACAGCGGGCTTGCGGATTTTGCGGATAAGATCCATTTTTATATGACAAACCCAACTGAACGTGAACGAATTGCGGCGCTTAATTATCAAGAGACAGTGCCGGCTTACACGTATGACCAGCTTCCGGCCGTGCTTACGGCTATTCGCTCCGATACTTTCTCA---\n",
      "\n",
      "Looking at: B_vallismortis\n",
      "Replacing with new seq:  ATGAACGTTTTATATATCCGTTCAGGATATAAAGGCATATACACGTATTTTGACCGCTGGATCGAGGAAGGCTTTTTACAGTCTTCCGTCCGCTTTTTTTCTGTTTCTGAGTTTGAGGAAGATACCATCCGCAGTATACGGGCATTCAAACCAGACATCAGTCTGATGATGACGGGAGACCGTATTCCGTCTGAATGGCTTCAATGGCTGAAGACAGAACAGATTCCCGTTTATCTATGGATGACGGAGGATCCTTTTTATTTCGATGTGAGCATGACAATCGCCCCTTTCGCCGATGCCGTTTTGACGATAGAACAAAACGCGCTTGATGCTTATCGGCAAATGGGCTTACAGCACGTATATTATGTACCGATCCCCGTTAATCAGCGGCTTTTTAAGAAACAACCTGCTGAAGCCTCTCTTCACACAAATCTGCTTCTGATCGGATACCCGTATCCGAACCGCGTTGAACTGGTGAAAGCAGCGGCAAAGCTTCCGTATACGCTTCGTATTATCGGGAAGGGATGGCGCAGACATCTGCCTAAAAAAATCGCCAGGCAGCAAAATGTGCTCGTAATGGATGAATGGATTGCACCCGAACGTGCCGCATTATATTACAACAGCGCAGACGTGGTCTTAAACCCTCACCGTTCTTATCGGTTTGCGCTCAATCAGAACGGACGCGGCATCGAAAATATCAGTTTAAATACAAGAAGCTTAGACATTGCGGCTTGTCAGGCCTTTCAGCTGACAGACCTTTCTCCCGCGCCCCCGTTTTCATCATTTGTTTCTTACAGCGGGTTTGCAGATTTTGCGGATAAGATCCAATTTTATATGACAAACCCGGGTGAACGTAAACAAATCGCGGCGCTTAATTACCAAGAGACAGTACCGGCGTACACGTATGACCGGCTGCCGGCTGTGCTTTCGGCTATCCGTTCCAAGACTTTCTCA---\n",
      "\n",
      "Looking at: B_velezensis\n",
      "Replacing with new seq:  ATGAACGTTTTATATATCCGTTCAGGATATAAAGGCATATACACGTATTTTGACCGCTGGATCGAGGAAGGCTTTTTACAGTCTTCCGTCCGCTTTTTTTCTGTTTCTGAGTTTGAGGAAGATACCATCCGCAGTATACGGGCATTCAAACCAGACATCAGTCTGATGATGACGGGAGACCGTATTCCGTCTGAATGGCTTCAATGGCTGAAGACAGAACAGATTCCCGTTTATCTATGGATGACGGAAGATCCTTTTTATTTCGATGTGAGCATGTCAATCGCCCCTTTCGCCGATGCCGTTTTAACGATAGAACAAAACGCGCTTGATGCTTACCGGCAAATGGGCTTACAGCACGTATATTATGTCCCGATCCCCGTTAATCAGCGGCTTTTTAAGAAACAACCTGCTGAAGCTTCTCTTCACACAAATCTGCTTCTGATCGGATACCCGTACCCGAACCGCGTTGAACTGGTGAAAGCAGCGGCGAAGCTTCCGTATACGCTTCGTATCATCGGGAAGGGATGGCGCAGACATCTGCCTAAAAAAATCGCCAGGCAGCAAAATGTGCTCGTAATGGATGAATGGATTGCACCCGAACGTGCCGCATTATATTACAACAGCGCAGACGTGGTCTTAAACCCTCACCGTTCTTATCGGTTTGCGCTCAATCAGAACGGACGCGGCATCGAAAATATCAGTTTAAATACGAGAAGCTTAGACATTGCGGCTTGTCAGGCCTTTCAGCTGACAGACCTTTCTCCCGCGCCCCCATTTTCATCCTTTGTTTCTTACAGCGGCCTTGCGGATTTTGCGGATAAGATCCATTTTTATATGACAAACCCGGGTGAACGTAAACAAATCGCGGCGCTTAATTACCAAGAGACAGTACCGGCGTACACGTATGACCGGCTGCCGGCTGTGCTTTCGGCTATCCGTTCCAAGACTTTCTCA---\n",
      "\n",
      "DOne here!\n",
      "\n",
      "Finallly DOne!\n"
     ]
    }
   ],
   "source": [
    "# Steps to creat a database of the same gene from multiple genomes/species (used for \"Selection methods\")\n",
    "\n",
    "#Step 2: Delete stop codons of a gene database v2\n",
    "\n",
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "\n",
    "codon_stop = [\"TAG\", \"TGA\", \"TAA\"]\n",
    "\n",
    "#list of multiple files\n",
    "input_files = [\"cgeA_aligned.fas\",\n",
    "              \"cgeB_aligned.fas\",\n",
    "              \"cgeC_aligned.fas\",\n",
    "              \"cgeD_aligned.fas\",\n",
    "              \"cgeE_aligned.fas\"]\n",
    "\n",
    "for file in input_files:    \n",
    "    handle = SeqIO.parse(file, \"fasta\")\n",
    "    print(\"Looking at: \" + file)\n",
    "    for record in handle: #open and iterate your file\n",
    "        print(\"Looking at: \" + record.id)\n",
    "        #print(record.seq)\n",
    "        tempRecordSeq = list(record.seq) #make the sequence a list (each letter is an element of the list)\n",
    "        for index in range(0, len(record.seq), 3): #iterate each time over 3 letters (codon) of the sequence\n",
    "            codon = record.seq[index:index+3] #recall that [i:i+3] is not inclusive, so it wil take from i to i+2\n",
    "            if codon in codon_stop: #check if the codon is a stop codon\n",
    "                tempRecordSeq[index:index+3] = '-','-','-' #replace each letter of the stop codon with \"-\" to not disturb the length of the alignment\n",
    "        record.seq = Seq(\"\".join(tempRecordSeq)) #definde a new sequence by joining the elements of the list\n",
    "        print(\"Replacing with new seq: \", record.seq + \"\\n\")\n",
    "        name = file.split(\".\")\n",
    "        output_file = open(name[0] + \"_nostop.fas\", \"a+\") #output file for the alignment\n",
    "        output_file.write(\">%s\\n%s\\n\" % (record.id, record.seq)) #write id + seq\n",
    "    print(\"DOne here!\" + \"\\n\")\n",
    "    handle.close()\n",
    "output_file.close()\n",
    "print(\"Finallly DOne!\")\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
